hydrus/hydraspec/openapi-hydra_parser.py

```diff
@@ -0,0 +1,113 @@
+"""
+Module to take in Open Api Specification and convert it to HYDRA Api Doc
+
+"""
+import yaml
+from doc_writer import HydraDoc, HydraClass, HydraClassProp, HydraClassOp
+import json
+
+with open("openapi_sample.yaml", 'r') as stream:
+    try:
+        doc = yaml.load(stream)
+    except yaml.YAMLError as exc:
+        print(exc)
+
+info = doc["info"]
+desc = info["description"]
+title = info["title"]
+baseURL = doc["host"]
+name = doc["basePath"]
+api_doc = HydraDoc(name, title, desc, name, baseURL)
+definitions = doc["definitions"]
```

Comment:
Definitions won't always be defined in doc["definitions"]. You should use `$ref` see this https://swagger.io/docs/specification/using-ref/

Reply:
yes you are right , right now i am first parsing definitions and adding props and classes to api doc , then i parse paths and check if the path has the same name as class , if yes i add the operations .
Would the following be a better implementation , parse the paths , when $ref is found , parse whatever is defined and create classes ?

Reply:
Yeah, you should check where $ref points and then extract the definitions from there. It can be in the same doc or at some remote location but for now using local definitions should be fine